nemoguardrails/llm/prompts/nemotron_reasoning.yml

```diff
@@ -0,0 +1,365 @@
+# Collection of all the prompts for Nemotron models
+# This file contains message-based prompts for Nemotron models
+prompts:
+  - task: general
+    models:
+      - nvidia/nemotron
+      - nemotron
+    messages:
+      # by default detailed thinking is off
+      # user can turn it on using the system message, not the other way
+      - type: system
+        content: |
+          {{ general_instructions }}{% if relevant_chunks != None and relevant_chunks != '' %}
+          This is some relevant context:
+          ```markdown
+          {{ relevant_chunks }}
+          ```{% endif %}
+      - "{{ history | to_chat_messages }}"
+
+  # Prompt for detecting the user message canonical form.
+  - task: generate_user_intent
+    models:
+      - nvidia/nemotron
+      - nemotron
+    messages:
+      - type: system
+        content: |
+          {{ general_instructions }}
+
+          Your task is to generate the user intent in a conversation given the last user message similar to the examples below.
+          Do not provide any explanations, just output the user intent.
+
+          # Examples:
+          {{ examples | verbose_v1 }}
+
+      - "{{ sample_conversation | first_turns(2) | to_messages }}"
+      - "{{ history | colang | to_messages }}"
+      - type: assistant
+        content: |
+          Bot thinking: potential user intents are: {{ potential_user_intents }}
+
+    output_parser: "verbose_v1"
+
+  # Prompt for generating the next steps.
+  - task: generate_next_steps
+    models:
+      - nvidia/nemotron
+      - nemotron
+    messages:
+      - type: system
+        content: |
+          {{ general_instructions }}
+
+          Your task is to generate the next steps in a conversation given the last user message similar to the examples below.
+          Do not provide any explanations, just output the user intent and the next steps.
+
+          # Examples:
+          {{ examples | remove_text_messages | verbose_v1 }}
+
+      - "{{ sample_conversation | first_turns(2) | to_intent_messages }}"
+      - "{{ history | colang | to_intent_messages }}"
+
+    output_parser: "verbose_v1"
+
+  # Prompt for generating the bot message from a canonical form.
+  - task: generate_bot_message
+    models:
+      - nvidia/nemotron
+      - nemotron
+    messages:
+      - type: system
+        content: |
+          detailed thinking on
```

Comment:
What happens with older non-reasoning nemotrons?

Reply:
For example, does https://build.nvidia.com/nvidia/llama-3_1-nemotron-70b-instruct match this model name pattern? If so, what will happen?

Reply:
thanks for realizing all these variations, I added them to the tests and also realized these are not affected by thinking on system message